app/connections/DataRelay.js

```diff
@@ -18,78 +18,227 @@
+                var parsingStr = stdout;
+                var match;
 
-    // First transmission is header columns
-    if (TelemetryData.getHeaders().length === 0) {
-      parseHeaders(data);
+                do {
+                    //search for any term in the form Bcast:#.#.#.# or broadcast:#.#.#.#
+                    match = parsingStr.toString().match(/(?:Bcast|broadcast):([\d.]*)/);
+
+                    //if match exists
+                    if (match != null) {
+                        //parse out any obviously wrong broadcasts (this could be improved)
+                        if (ip.isV4Format(match[1])) {
+                            //Connect to udp using given broadcastIP
+                            connectUDP(match[1]);
+                        }
+
+                        parsingStr = parsingStr.substring(match.index + 1);
+                    }
+                } while (match != null);
+            }
+        });
     }
-    else { //if its the non-header columns(actual data)
-      parseData(data);
+}
+
+
+/**
+ * Connects to TCP on given host and port
+ * @connectTCP
+ */
+var connectTCP = function(host,port) {
+  //remove all previous data-relay connections
+    if (NetworkManager.getConnectionByName('data_relay')) {
+        NetworkManager.removeAllConnections('data_relay');
     }
-  }.bind(this));
-};
+    var data_relay = NetworkManager.addConnection('data_relay', host, port);
+
+    data_relay.setTimeout(network_config.get('datarelay_timeout'));
+
+    data_relay.on('connect', function() {
+        TelemetryData.clearHeaders();
+        StatusManager.setStatusCode('CONNECTED_DATA_RELAY', true);
+    });
+
+    data_relay.on('close', function(had_error) {
+        StatusManager.setStatusCode('DISCONNECTED_DATA_RELAY', true);
+    });
+
+    data_relay.on('timeout', function() {
+        StatusManager.setStatusCode('TIMEOUT_DATA_RELAY', true);
+    });
+
+    data_relay.on('write', function(data) {
+        StatusManager.addStatus('Sent command to data_relay', 3, 2000);
+    });
+
+    data_relay.on('data', function(data) {
+        if (!data) { //don't do anything if we get blank data or anything that's not an object
+            Logger.error('Got a blank packet from the data relay station. Value: ' + data);
+            return;
+        }
+
+        data = data.toString();
+
+        // First transmission is header columns
+        if (TelemetryData.getHeaders().length === 0) {
+            parseHeaders(data);
+        } else { //if its the non-header columns(actual data)
+            parseData(data);
+        }
+    }.bind(this));
+}
+
+
+/**
+ * connect to data-relay using UDP broadcast address
+ * @connectUDP
+ */
+var connectUDP = function(broadcastIP){
+    var dgram = require('dgram');
+    var server = dgram.createSocket('udp4');
+    var port = network_config.get('datarelay_port');
+
+    var udp_open = false;
+
+    server.on('error', (err) => {
```

Comment:
I'd like to keep the codebase ES5 consistent, so dont use arrow functions (I do use the `${variable}` in my console logs however, since its an amazing feature)